### PR TITLE
osd: drop unnecessary transaction cleanup

### PR DESCRIPTION
**Pull request by xiexingguo**
The lock() method will ensure 'dirty_info' will be false,
and within this function's scope I see no possibility to
set 'dirty_info' true. Thus I guess the transaction cleanup
logic before exit is never reachable and therefore shall be
considered as redundant and can be safely removed.

Signed-off-by: xie xingguo <xie.xingguo@zte.com.cn>

### CODE REVIEW

**Comment by xiexingguo** [comment]
@liewegas Have a look?

**Comment by xiexingguo** [comment]
@tchaikov You might want to review this one too, since you are there :-)

File: src/osd/PG.cc [hunk 1]
```diff
@@ -2008,13 +2008,6 @@ void PG::_activate_committed(epoch_t epoch, epoch_t activation_epoch)
     }
   }
 
-  if (dirty_info) {
```
Comment (asheplyakov) [inline src/osd/PG.cc]: What about
`assert(!dirty_info);`

Comment (xiexingguo) [inline src/osd/PG.cc]: good advice, thank you, sir!

**Pull request closed without merging**